rust/nasl-interpreter/src/built_in_functions/misc.rs

```diff
@@ -154,6 +182,128 @@ pub fn gunzip(_: &str, _: &dyn Sink, register: &Register) -> Result<NaslValue, F
         }
     }
 }
+/// Takes seven named arguments sec, min, hour, mday, mon, year, isdst and returns the Unix time.
+pub fn mktime(_: &str, _: &dyn Sink, register: &Register) -> Result<NaslValue, FunctionError> {
+
+    let sec = match register.named("sec") {
+        Some(ContextType::Value(NaslValue::Number(x))) => *x as u32,
+        _ => 0,
+    };
+    let min = match register.named("min") {
+        Some(ContextType::Value(NaslValue::Number(x))) => *x as u32,
+        _ => 0,
+    };
+    let hour = match register.named("hour") {
+        Some(ContextType::Value(NaslValue::Number(x))) =>  *x as u32,
+        _ => 0,
+    };
+    let mday = match register.named("mday") {
+        Some(ContextType::Value(NaslValue::Number(x))) => *x as u32,
+        _ => 0,
+    };
+    let mon = match register.named("mon") {
+        Some(ContextType::Value(NaslValue::Number(x))) => *x as u32,
+        _ => 1,
+    };
+    let year = match register.named("year") {
+        Some(ContextType::Value(NaslValue::Number(x))) => *x as i32,
+        _ => 0,
+    };
+
+    // TODO: fix isdst
+    let isdst = match register.named("isdst") {
+        Some(ContextType::Value(NaslValue::Number(x))) => *x as i32,
+        _ => -1,
+    };
+
+    let offset = chrono::Local::now().offset().fix().local_minus_utc();
+    let r_dt = Utc.with_ymd_and_hms(year, mon, mday, hour, min, sec);
+    match r_dt {
+        LocalResult::Single(x) => Ok(NaslValue::Number(x.naive_local().timestamp() - offset as i64)),
+        _ => Ok(NaslValue::Null),
+    }
+}
+
+
+
+/// Returns an dict(mday, mon, min, wday, sec, yday, isdst, year, hour) based on optional given time in seconds and optional flag if utc or not.
+pub fn localtime(_: &str, _: &dyn Sink, register: &Register) -> Result<NaslValue, FunctionError> {
```

Comment:
This whole function seems to be unnecessary hard to read and I get a lot of clippy warnings.

Reply:
After looking into it, I would rather rewrite it to:

```
fn create_localtime_map<T>(date: chrono::DateTime<T>) -> HashMap<String, NaslValue>
where
    T: chrono::TimeZone,
{
    HashMap::from([
        ("sec".to_string(), NaslValue::from(date.second() as i64)),
        ("min".to_string(), NaslValue::from(date.minute() as i64)),
        ("hour".to_string(), NaslValue::from(date.hour() as i64)),
        ("mday".to_string(), NaslValue::from(date.day() as i64)),
        ("mon".to_string(), NaslValue::from(date.month() as i64)),
        ("year".to_string(), NaslValue::from(date.year() as i64)),
        ("wday".to_string(), NaslValue::from(date.weekday() as i64 + 1)),
        ("yday".to_string(), NaslValue::from(date.ordinal() as i64)),
        // TODO: fix isdst
        ("isdst".to_string(), NaslValue::from(0)),
    ])
}

/// Returns an dict(mday, mon, min, wday, sec, yday, isdst, year, hour) based on optional given time in seconds and optional flag if utc or not.
pub fn localtime(_: &str, _: &dyn Sink, register: &Register) -> Result<NaslValue, FunctionError> {
    let utc_flag = match register.named("utc") {
        Some(ContextType::Value(NaslValue::Number(x))) => *x != 0,
        Some(ContextType::Value(NaslValue::Boolean(x))) => *x,
        _ => false,
    };

    let secs = match register.positional() {
        [] => 0,
        [x0, ..] => i64::from(x0),
    };
    let date = match (utc_flag, secs) {
        (true, 0) => create_localtime_map(Utc::now()),
        (true, secs) => match Utc.timestamp_opt(secs, 0) {
                LocalResult::Single(x) => create_localtime_map(x),
            _ => create_localtime_map(Utc::now()),
        }
        (false, 0) => create_localtime_map(Local::now()),
        (false, secs) => match Local.timestamp_opt(secs, 0) {
            LocalResult::Single(x) => create_localtime_map(x),
            _ => create_localtime_map(Local::now())
        },
    };

    Ok(NaslValue::Dict(date))
}
```

Reply:
Feel free add a new commit with your solution.